tensorboard/webapp/metrics/views/card_renderer/scalar_card_data_table.ts

```diff
@@ -183,7 +196,22 @@ export class ScalarCardDataTable {
         }
         return selectedStepData;
       });
+    dataTableData.sort(
+      (point1: SelectedStepRunData, point2: SelectedStepRunData) => {
+        if (
+          point1[this.sortingInfo.header]! < point2[this.sortingInfo.header]!
+        ) {
+          return this.sortingInfo.order === SortingOrder.ASCENDING ? -1 : 1;
+        }
+        if (
+          point1[this.sortingInfo.header]! > point2[this.sortingInfo.header]!
+        ) {
+          return this.sortingInfo.order === SortingOrder.ASCENDING ? 1 : -1;
+        }
 
+        return 0;
```

Comment:
nitpick: This if -> ternary could really just be a ternary. Sorting is always a little funny so I MIGHT have these values backwards.
```suggestion
        return this.sortingInfo.order === SortingOrder.ASCENDING ? point2[this.sortingInfo.header]! - point1[this.sortingInfo.header]! : point1[this.sortingInfo.header]! - point2[this.sortingInfo.header]!;
```

Reply:
For posterity: As we discussed the problem here is that this is sometimes text and we cannot subtract it.